From 3451189a3f11c9d1af65e171ebc512ab94e99187 Mon Sep 17 00:00:00 2001
From: KO Myung-Hun <komh@chollian.net>
Date: Tue, 2 Jan 2018 18:03:35 +0900
Subject: [PATCH 21/26] Revert "qt: check if QDesktopWidget::screenNumber()
 returns a valid screen"

This reverts commit fbf1f129ce7ec1387c9ead843c01c243337be460.
---
 modules/gui/qt/components/controller.cpp |  3 ---
 modules/gui/qt/main_interface.cpp        | 33 ++++++++++++++------------------
 2 files changed, 14 insertions(+), 22 deletions(-)

diff --git a/modules/gui/qt/components/controller.cpp b/modules/gui/qt/components/controller.cpp
index 37d0d2cc4e..11f74b13a7 100644
--- a/modules/gui/qt/components/controller.cpp
+++ b/modules/gui/qt/components/controller.cpp
@@ -869,9 +869,6 @@ void FullscreenControllerWidget::restoreFSC()
         setMinimumWidth( FSC_WIDTH );
         adjustSize();
 
-        if ( targetScreen() < 0 )
-            return;
-
         QRect currentRes = QApplication::desktop()->screenGeometry( targetScreen() );
         windowHandle()->setScreen(QGuiApplication::screens()[targetScreen()]);
 
diff --git a/modules/gui/qt/main_interface.cpp b/modules/gui/qt/main_interface.cpp
index 72de45d7d7..fd8fc6eb55 100644
--- a/modules/gui/qt/main_interface.cpp
+++ b/modules/gui/qt/main_interface.cpp
@@ -870,28 +870,23 @@ void MainInterface::setVideoFullScreen( bool fs )
             numscreen = QApplication::desktop()->screenNumber( p_intf->p_sys->p_mi );
 
         fullscreenControls->setTargetScreen( numscreen );
+        QRect screenres = QApplication::desktop()->screenGeometry( numscreen );
+        lastWinScreen = windowHandle()->screen();
+        windowHandle()->setScreen(QGuiApplication::screens()[numscreen]);
 
-        if ( numscreen >= 0 )
+        /* To be sure window is on proper-screen in xinerama */
+        if( !screenres.contains( pos() ) )
         {
+            lastWinPosition = pos();
+            lastWinSize = size();
+            msg_Dbg( p_intf, "Moving video to correct position");
+            move( QPoint( screenres.x(), screenres.y() ) );
+        }
 
-            QRect screenres = QApplication::desktop()->screenGeometry( numscreen );
-            lastWinScreen = windowHandle()->screen();
-            windowHandle()->setScreen(QGuiApplication::screens()[numscreen]);
-
-            /* To be sure window is on proper-screen in xinerama */
-            if( !screenres.contains( pos() ) )
-            {
-                lastWinPosition = pos();
-                lastWinSize = size();
-                msg_Dbg( p_intf, "Moving video to correct position");
-                move( QPoint( screenres.x(), screenres.y() ) );
-            }
-
-            /* */
-            if( playlistWidget != NULL && playlistWidget->artContainer->currentWidget() == videoWidget )
-            {
-                showTab( videoWidget );
-            }
+        /* */
+        if( playlistWidget != NULL && playlistWidget->artContainer->currentWidget() == videoWidget )
+        {
+            showTab( videoWidget );
         }
 
         /* we won't be able to get its windowed sized once in fullscreen, so update it now */
-- 
2.13.3

